### PR TITLE
redux added to the app

### PR DESCRIPTION
**Pull request by MansiM578**

### CODE REVIEW

File: src/components/header/Header.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  marginLeft: 4,
+});
+
+const CustomButtonWrapper = styled(Box)`
+  margin: 0 5% 0 auto;
+`;
+
+export default function Header() {
+  const logoURL =
+    "https://static-assets-web.flixcart.com/www/linchpin/fk-cp-zion/img/flipkart-plus_8d85f4.png";
```
Comment (amishaaggarwal) [inline src/components/header/Header.tsx:39]: use static files instead

File: src/components/header/Header.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  const subURL =
+    "https://static-assets-web.flixcart.com/www/linchpin/fk-cp-zion/img/plus_aef861.png";
```
Comment (amishaaggarwal) [inline src/components/header/Header.tsx:41]: use static files instead

File: src/components/listItems/ProductList.tsx [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+`;
+
+const Model = styled(Box)`
+  padding: "25px 15px";
+  text-align: center;
+  background-color: #ffffff;
+`;
+
+const ProductList: React.FC = () => {
+  // const navigate = useNavigate();
```
Comment (amishaaggarwal) [inline src/components/listItems/ProductList.tsx]: remove unused code

File: src/components/listItems/ProductList.tsx [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    return <CircularProgress />;
+  }
+
+  if (error) {
+    return <Typography color="error">{error}</Typography>;
+  }
+  return (
+    // <Box>
+    //   {items.map((item) => (
+    //     <div key={item.id}>
```
Comment (amishaaggarwal) [inline src/components/listItems/ProductList.tsx]: remove unused code

File: src/components/listItems/ProductList.tsx [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+          spacing={{ xs: 2, md: 3 }}
+          columns={{ xs: 4, sm: 8, md: 12, lg: 12 }}
+        >
+          {items?.map((item) => (
+            <Grid item xs={2} sm={2} md={2} key={item.id}>
+              <Model textAlign="center" style={{ padding: "25px 15px" }}>
+                <Image src={item.images[0]} alt="phone1" />
+                <Link>{item.name}</Link>
+                {/* <Link
+                  onClick={() => {
```
Comment (amishaaggarwal) [inline src/components/listItems/ProductList.tsx]: remove unused code

File: src/reducers/Store.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+export default store;
+
+// import { configureStore } from "@reduxjs/toolkit";
+// import thunk from "redux-thunk";
+// import logger from "redux-logger";
+// import rootReducer from "components/listItems/RootReducer";
+// import { TypedUseSelectorHook, useDispatch, useSelector } from "react-redux";
+
+// const store = configureStore({
+//   reducer: rootReducer,
```
Comment (amishaaggarwal) [inline src/reducers/Store.ts]: remove unused code

File: src/reducers/ProductSlice.ts [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+      state.loading = false;
+      state.items = null;
+      state.error = action.error.message || "An error occurred.";
+    },
+  },
+});
+
+export default itemsSlice.reducer;
+
+export const selectItems = (state: RootState) => state.items.items;
```
Comment (amishaaggarwal) [inline src/reducers/ProductSlice.ts:187]: remove these functions if not being used

**Review by amishaaggarwal — CHANGES_REQUESTED** [review]

File: src/components/details/ProductDetail.tsx [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+          </ColumnText>
+          <ColumnText>
+            <TableCell style={{ color: "#878787" }}>Warranty</TableCell>
+            <TableCell>No Warranty</TableCell>
+          </ColumnText>
+          <ColumnText>
+            <TableCell style={{ color: "#878787" }}>Seller</TableCell>
+            <TableCell>
+              <Box component="span" style={{ color: "#2874f0" }}>
+                {item?.seller[0].name}
```
Comment (amishaaggarwal) [inline src/components/details/ProductDetail.tsx]: apply optional chaining throughout eg item?.seller?.[0]?.name

File: src/components/details/actionItem.tsx [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
...X47FPEYpQ2fpKrOe665ma6Ef7cfm9re3mzpYnPSoO+ulOcVtve1+nkjgY8zT+rNUJu075e36+ZrNY9eY6PE01vTTr8u0FTv8W27+ZVkzxW8YpKnF0lu3s/bRVPK2ncnvvvu7RTiV36N/r0FZGbLfuE7z3UEY3v4WLOk6MtRObK5c/D6/U7HZ2XViS6x/D6unuOLKUaStN308D0HZyisSUd+dvlbNeP6z7+YmvrcRxNLrxKpeT/Q6uLNLGI8ZfISyCl4yHEusLAzaSGjQyKAu4biTS+IficbHM0RzkrbrLiLVGTStRlXEvon69g1zb6L0vf3CpEdo4VJ3Ryp8FPpCTXode06rl4tiT4h9L2M+szW/PrfLjTi1tJNdd00Niy0dOOeVN1surW1+HIZP8sp4k03Wytvyb6GOXTu/wBxz1mV+T5+KK1kadx2O5HsyGRtrFkikucacb36O/I5eTh5QtSg9qt0635bjc0zc/GdSdVez5kdyy1IulToQ31GVYzvdkqsb9Jy4wOnwjqJvxk1z9+q1tiSb8ityEb8zo5JYtiykI5EFjkRZXqDUA7kRqF1EWBXDEi+GPwK1kSJ+9UqTftZh0aVhVb0VycV4mZ52+bIWTz6XvsKLsuXqk0jHkk+f7Fs+IVcl5+JlyZL/l/UzutZgjkpl8ON2qvy+ncyuvn0GVVXpJWtzHTXajpaZOK2Uop2vSvf6BuH4pSuDr8dxb6tPan08PYctR2Jr6/Qt1Jh1DoPGJCGTNYxp4xNMZ0jIpDazTOtDmI5lWsjUKkWaiNQmoixVizURqEsLBFlkWJZGoUil5BXlMveEORx6ejhplloWWUoslErXOLYvr09g01S8mUJja9q9YpD11T9RIsJVb9RZGS9/uKzp4S3T9oz50Uxe9Fmven0KzuLHLZPryZGsnLJKL3V+HXp7tzOplqc1fqJ1FKkGotTlbqJ1FWoNQpFmolSKrJsUizUFldhYqRZqCyuwsVYwgAHJ6AhkxQAZsiyAAsi9hsT6lVjp7e8JuLJun6P1Ijlp/XvK5ysQtTPK3JO2QmJYWSrFqkTZUmTZakW2Goqsmy1It1BZVqJ1CkW6gsr1BqLUiywsrsLFIpAAMOgAAAAAAAAAAAAAAAAALAAJsLIACbJsUAGsLFAB7CxCbFIgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA//Z"
```
Comment (amishaaggarwal) [inline src/components/details/actionItem.tsx]: remove comments

File: src/components/header/Header.tsx [hunk 10]
```diff
@@ -36,7 +50,10 @@ export default function Header() {
   return (
     <StyledHeader>
       <Toolbar style={{ minHeight: 55 }}>
-        <Component>
+        {/* <IconButton>
```
Comment (amishaaggarwal) [inline src/components/header/Header.tsx]: remove comments

**Review by amishaaggarwal — CHANGES_REQUESTED** [review]

File: src/components/listItems/ProductList.tsx (hunk 3 above)
Comment (MansiM578) [reply src/components/listItems/ProductList.tsx]: Done

File: src/components/listItems/ProductList.tsx (hunk 4 above)
Comment (MansiM578) [reply src/components/listItems/ProductList.tsx]: Done

File: src/components/listItems/ProductList.tsx (hunk 5 above)
Comment (MansiM578) [reply src/components/listItems/ProductList.tsx]: Done

File: src/reducers/Store.ts (hunk 6 above)
Comment (MansiM578) [reply src/reducers/Store.ts]: Done

File: src/reducers/ProductSlice.ts (hunk 7 above)
Comment (MansiM578) [reply src/reducers/ProductSlice.ts:187]: Used

File: src/components/details/ProductDetail.tsx (hunk 8 above)
Comment (MansiM578) [reply src/components/details/ProductDetail.tsx]: Done

File: src/components/details/actionItem.tsx (hunk 9 above)
Comment (MansiM578) [reply src/components/details/actionItem.tsx]: Done

File: src/components/header/Header.tsx (hunk 10 above)
Comment (MansiM578) [reply src/components/header/Header.tsx]: Done

**Pull request merged**